scripts/models.py

```diff
@@ -0,0 +1,244 @@
+from __future__ import annotations
+from enum import Enum
+from pydantic import BaseModel, Field, AnyUrl, EmailStr
+from typing import List, Union, Optional
+from datetime import date
+
+
+class AccessType(str, Enum):
+    Open = "Open"
+    Embargoed = "Embargoed"
+    Restricted = "Restricted"
+
+
+class License(str, Enum):
+    cc0 = "CC0-1.0"
+    ccby = "CC-BY-4.0"
+    ccbync = "CC-BY-NC-4.0"
+    pddl = "PDDL-1.0"
+
+
+class RoleType(str, Enum):
+    Author = "Author"
+    Conceptualization = "Conceptualization"
+    ContactPerson = "ContactPerson"
+    DataCollector = "DataCollector"
+    DataCurator = "DataCurator"
+    DataManager = "DataManager"
+    FormalAnalysis = "FormalAnalysis"
+    FundingAcquisition = "FundingAcquisition"
+    Investigation = "Investigation"
+    Maintainer = "Maintainer"
+    Methodology = "Methodology"
+    Producer = "Producer"
+    ProjectLeader = "ProjectLeader"
+    ProjectManager = "ProjectManager"
+    ProjectMember = "ProjectMember"
+    ProjectAdministration = "ProjectAdministration"
+    Researcher = "Researcher"
+    Resources = "Resources"
+    Software = "Software"
+    Supervision = "Supervision"
+    Validation = "Validation"
+    Visualization = "Visualization"
+    Funder = "Funder"
+    Sponsor = "Sponsor"
+    StudyParticipant = "StudyParticipant"
+    Other = "Other"
+
+
+class Relation(str, Enum):
+    IsCitedBy = "IsCitedBy"
+    Cites = "Cites"
+    IsSupplementTo = "IsSupplementTo"
+    IsSupplementedBy = "IsSupplementedBy"
+    IsContinuedBy = "IsContinuedBy"
+    Continues = "Continues"
+    Describes = "Describes"
+    IsDescribedBy = "IsDescribedBy"
+    HasMetadata = "HasMetadata"
+    IsMetadataFor = "IsMetadataFor"
+    HasVersion = "HasVersion"
+    IsVersionOf = "IsVersionOf"
+    IsNewVersionOf = "IsNewVersionOf"
+    IsPreviousVersionOf = "IsPreviousVersionOf"
+    IsPartOf = "IsPartOf"
+    HasPart = "HasPart"
+    IsReferencedBy = "IsReferencedBy"
+    References = "References"
+    IsDocumentedBy = "IsDocumentedBy"
+    Documents = "Documents"
+    IsCompiledBy = "IsCompiledBy"
+    Compiles = "Compiles"
+    IsVariantFormOf = "IsVariantFormOf"
+    IsOriginalFormOf = "IsOriginalFormOf"
+    IsIdenticalTo = "IsIdenticalTo"
+    IsReviewedBy = "IsReviewedBy"
+    Reviews = "Reviews"
+    IsDerivedFrom = "IsDerivedFrom"
+    IsSourceOf = "IsSourceOf"
+    IsRequiredBy = "IsRequiredBy"
+    Requires = "Requires"
+    Obsoletes = "Obsoletes"
+    IsObsoletedBy = "IsObsoletedBy"
+
+
+class IdentifierType(str, Enum):
+    doi = "DOI"
+    orcid = "ORCID"
+    ror = "ROR"
+    dandi = "DANDI"
+    rrid = "RRID"
+    none = "No identifier prefix"
+
+
+class PropertyValue(BaseModel):
+    maxValue: float = None
+    minValue: float = None
+    unitCode: Union[str, AnyUrl] = None
+    unitText: str = None
+    value: Union[bool, float, str, int, List[Union[bool, float, str, int]]] = None
+    valueReference: PropertyValue = None
```

Comment:
@satra, could you explain what `valueReference` is, and why it references an instance of `PropertyValue`?

Reply:
https://schema.org/valueReference

Reply:
as an example if some measures intensity/pressure, one can measure it with respect to a reference measurement, and valueReference allows that.

Reply:
Okay. This causes an error on the frontend, as the schema is circular here. I've so far dealt with it manually. If this needs to remain this way, I can simply keep that in place.